packages/cubejs-databricks-jdbc-driver/src/DatabricksDriver.ts

```diff
@@ -669,19 +702,30 @@ export class DatabricksDriver extends JDBCDriver {
     const blobsList = containerClient.listBlobsFlat({ prefix: foldername });
     for await (const blob of blobsList) {
       if (blob.name && expr.test(blob.name)) {
-        const sas = generateBlobSASQueryParameters(
-          {
-            containerName: container,
-            blobName: blob.name,
-            permissions: ContainerSASPermissions.parse('r'),
-            startsOn: new Date(new Date().valueOf()),
-            expiresOn:
-              new Date(new Date().valueOf() + 1000 * 60 * 60),
-            protocol: SASProtocol.Https,
-            version: '2020-08-04',
-          },
-          credential,
-        ).toString();
+        const startsOnDate = new Date(new Date().valueOf());
+        const expiresOnDate = new Date(new Date().valueOf() + 1000 * 60 * 60);
+        const signatureValues: BlobSASSignatureValues = {
+          containerName: container,
+          blobName: blob.name,
+          permissions: ContainerSASPermissions.parse('r'),
+          startsOn: startsOnDate,
+          expiresOn: expiresOnDate,
+          protocol: SASProtocol.Https,
+          version: '2020-08-04',
+        };
+        const sas = credential instanceof StorageSharedKeyCredential
```

Comment:
`StorageSharedKeyCredential` should still be default

Reply:
@paveltiunov `generateBlobSASQueryParameters(signatureValues, credential)` only accept the credential with the type StorageSharedKeyCredential, but we may have a ClientSecretCredential one. Therefore, making StorageSharedKeyCredential as default here will have an error. Thanks.

Reply:
@MaggieZhang-01 I guess to make it more obvious to read and digest let's then make exhaustive match instead. Like
```
if (credential instanceof StorageSharedKeyCredential) {
  // ...
} else if (credential instanceof ClientSecretCredential) {
  // ...
} else {
  throw new Error(`Unexpected credentials type: ${credential}`);
}
```
Please make sure credential has a `toString()` implemented so it's properly rendered within an Error.

Reply:
@paveltiunov Many thanks for your suggestions, I added the error message with the type of credential in the last commit. I will make changes if this is not okay.

Reply:
@paveltiunov Would you be willing to help review this PR, I have applied the changes based on your previous review, thanks.